paasta_tools/deployd/watchers.py

```diff
@@ -242,7 +243,11 @@ def get_service_instances_with_changed_id(marathon_client, instances, cluster):
                                               instance=instance,
                                               cluster=cluster,
                                               soa_dir=DEFAULT_SOA_DIR)
-        if config.format_marathon_app_dict()['id'] not in marathon_app_ids:
+        try:
+            config_app_id = config.format_marathon_app_dict()['id']
+        except NoDockerImageError:
+            config_app_id = None
+        if not config_app_id or config_app_id not in marathon_app_ids:
```

Comment:
Hmm, I guess the brackets don't matter in this case because the unit test passes. It'd be clearer with them though so I'll send a fixup.

Regarding draining apps etc. we only call this code path from the config file watchers. So the watchers that are watching for autoscaling and maintenance don't check the app id and just put the service straight into the queue.